app/controllers/api/base_controller.rb

```diff
@@ -260,6 +244,68 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map { |key, value| extract_resource_from_param(key, value) }
+
+      parent_name, parent_class, parent_id = classes_arr.find{|kvp| not kvp[1].nil?}
```

Comment:
nitpick - formatting spacebars and readability
```
parent_name, parent_class, parent_id = classes_arr.find { |_, klass, _| klass.present? }
```
I suppose klass can't be empty string... if you prefer using negative condition, I find `!klass.nil?` better than `not klass.nil?`